alg/gdaltransformer.cpp

```diff
@@ -2011,13 +2015,22 @@ GDALCreateGenImgProjTransformer2( GDALDatasetH hSrcDS, GDALDatasetH hDstDS,
 
     else if( pszMethod != nullptr )
     {
-        CPLError(CE_Failure, CPLE_AppDefined,
-                 "Unable to compute a %s based transformation between "
-                 "pixel/line and georeferenced coordinates for %s.",
-                 pszMethod, GDALGetDescription(hSrcDS));
-
-        GDALDestroyGenImgProjTransformer( psInfo );
-        return nullptr;
+        // Try to instantiate as a registered transformatio method.
+        // What about passing transformer options? 
+        psInfo->pSrcTransformArg =
+            GDALCreateTransformer(pszMethod, hSrcDS, "SRC_", papszOptions);
```

Comment:
```suggestion
            GDALCreateTransformer(pszMethod, "SRC_", papszOptions);
```

Reply:
My premise is that if we pass the source dataset, the transformer creator can reach into it to pull supporting metadata instead of requiring they all be passed explicitly in options.

Reply:
OK, I was just reflecting that the current GDALCreateTransformer() implementation doesn't use hSrcDS

Reply:
I do not see any function called GDALCreateTransformer() in master - did you just mean similar ones, or the instantiation from XML?

Reply:
by "current" I meant your current implementation in this PR.